llvm/lib/Linker/LinkModules.cpp

```diff
@@ -473,6 +474,9 @@ bool ModuleLinker::run() {
       return true;
     ComdatsChosen[&C] = std::make_pair(SK, From);
 
+    if (From == LinkFrom::Dst)
```

Comment:
I think for COFF (not for ELF) in the comdat-refer-to-discarded.ll test case we should get a result of `LinkFrom::Both` here.

Reply:
If I remember it right, `LinkFrom::Both` corresponds to `comdat nodeduplicate`, while `comdat-refer-to-discarded.ll` uses `comdat any`. Do you think the interpretation of a comdat selection type should depend on the target platform (COFF vs ELF)?